### ISSUE TITLE
model: properties always appear to come from the top-level binding file

### ISSUE THREAD

**Issue by dottspina** [dottspina/dtsh#5]
When using the `cat` command to show the specifications of node properties, they all appear to be defined (actually *last modified*) by the node bindings itself, not the YAML file we would expect: e.g. all properties of " nordic,nrf52-flash-controller" devices appear to be defined by `nordic,nrf52-flash-controller.yaml`, although we should also find `flash-controller.yaml`, `base.yaml` and `pm.yaml`.         

An example is the `wakeup-source` property, which is defined in `pm.yaml`:

```
/
> cat &flash_controller$wakeup-source -Bl
           Name: wakeup-source                     
           From: nordic,nrf52-flash-controller.yaml
           Type: boolean                           
       Required: No                                
     Deprecated: No                                
           Enum: Not an enum                       
          Const: Not a const                       
        Default: No default value                  
Specifier Space: No specifier space    
```

This is a known issue in Zephyr's Python devicetree library: [edtlib: PropertySpec.path API might be incorrect or misleading #65135](https://github.com/zephyrproject-rtos/zephyr/issues/65135)

A pull request has been submitted upstream: [ edtlib: fix last modified semantic in included property specs #65221](https://github.com/zephyrproject-rtos/zephyr/pull/65221) 

While this bug has absolutely no incidence on Zephyr's use of the library, this is an issue for DTSh which aims to help beginners understand Devicetree: a tool that says that the `wakeup-source` property is defined by the bindings for the "nordic,nrf52-flash-controller" compatible does not.

Since we already have to repackage Zephyr's Python devicetree library (#2, [#5e803eb](https://github.com/dottspina/dtsh/commit/5e803eb)), should we consider applying the above patch ?

**Comment by dottspina** [comment]
Fixed upstream, patch merged.

**Comment by dottspina** [comment]
The patch didn't work for child-bidings, and introduced regressions.
We decided to revert it upstream ([edtlib: Revert a regression affecting property filters at the child-binding level and bellow #80030](https://github.com/zephyrproject-rtos/zephyr/pull/80030)).

DTSh now implements a workaround, where it retrieves properties lineage *a posteriori*.
Starting from the "top-level" binding, down to the recursively included YAML files, and honoring filters, we assume:
- the first binding file that adds some definition (e.g. `required: true`) to a property specification, is where the property was last modified
- the file where we find a description for the property is where it's initially specified

I'm reopening this until our workaround proves correct.

**Comment by dottspina** [comment]
Closing this: the workaround has been released with DTSh v0.2.4.
Incorrect properties lineage would be a different issue.
